### PR TITLE
Add "skipIfEqualOrNewerInstalled" command line switch...

### PR DESCRIPTION
**Pull request by KevinH-MS**
...and return unique exit codes for each type of failure.

### CODE REVIEW

**Comment by KevinH-MS** [comment]
FYI @tannergooding 
/CC @dotnet/roslyn-infrastructure 

Background:  For the LUT repo, I want to be able to call VsixExpInstaller to deploy a matching or newer version of Roslyn than what I build against, but short-circuit (with a message) if the version installed is equal or newer than what I already have.

**Comment by KevinH-MS** [comment]
FYI @tannergooding I've updated this to also do the "up to date" check on uninstall.  Please take a look.

**Comment by KevinH-MS** [comment]
Ping for review...

File: src/VSIXExpInstaller/VSIXExpInstaller/Program.cs [hunk 1]
```diff
@@ -49,14 +57,15 @@ private static bool FindArg(List<string> args, string argName)
...
-            Console.WriteLine("Usage: VSIXExpInstaller [/rootSuffix:suffix] [/u] [/uninstallAll] [/vsInstallDir:vsInstallDir] <vsix>");
+            Console.WriteLine("Usage: VSIXExpInstaller [/rootSuffix:suffix] [/skipIfEqualOrNewerInstalled] [/u] [/uninstallAll] [/vsInstallDir:vsInstallDir] <vsix>");
             Console.WriteLine("Suffix is Exp by default.");
         }
 
         private static string GetDevenvPath(string vsInstallDir)
         {
             if (string.IsNullOrWhiteSpace(vsInstallDir))
             {
+                Environment.ExitCode = E_ACCESSDENIED;
```
Comment (tannergooding) [inline src/VSIXExpInstaller/VSIXExpInstaller/Program.cs]: nit: I thought you could set the ErrorCode of an exception and it would do the same...

Comment (KevinH-MS) [inline src/VSIXExpInstaller/VSIXExpInstaller/Program.cs]: Didn't know that.  I will try.

Comment (KevinH-MS) [inline src/VSIXExpInstaller/VSIXExpInstaller/Program.cs]: It doesn't *seem* to work that way.  I tried throwing a Win32Exception and set ErrorCode, but the ERRORLEVEL was still -532462766.

**Pull request merged**